services/apigateway/route.go

```diff
@@ -78,9 +138,64 @@ func (c *Client) SetApiGatewayRouteRead(d *schema.ResourceData, route apigateway
 		}
 
 		if err := d.Set("upstreams", upstreams); err != nil {
-			return fmt.Errorf("error setting upstreams for the ApiGateway route with ID: %s", *route.Id)
+			return fmt.Errorf("error setting upstreams for the API Gateway Route with ID: %s", *route.Id)
 		}
 	}
 
 	return nil
 }
+
+func setRoutePostRequest(d *schema.ResourceData) *apigateway.RouteCreate {
+	return apigateway.NewRouteCreate(*setRouteConfig(d))
+}
+
+func setRoutePutRequest(d *schema.ResourceData) *apigateway.RouteEnsure {
+	routeID := d.Id()
+	route := setRouteConfig(d)
+
+	return apigateway.NewRouteEnsure(routeID, *route)
+}
+
+func setRouteConfig(d *schema.ResourceData) *apigateway.Route {
+	routeName := d.Get("name").(string)
+	routeType := d.Get("type").(string)
+	pathsRaw := d.Get("paths").([]interface{})
+	methodsRaw := d.Get("methods").([]interface{})
+	websocket := d.Get("websocket").(bool)
+	upstreamsRaw := d.Get("upstreams").([]interface{})
+
+	paths := make([]string, 0)
```

Comment:
just a nitpick about these `make([]string, 0)` declarations,

If you insist in calling a `make` beforehand we should probably do `paths := make([]string, len(pathsRaw))`, otherwise simply remove the `make` call since it's redundant with the `appends` in the for loop

Reply:
Fair, I'll change it to `make([]string, len(pathsRaw))`. This was more trying to figure out if this was what the linter wanted :))))